File: backend/src/main/java/com/lacrisalide/controller/EmailController.java
package com.lacrisalide.controller;

import com.lacrisalide.dto.email.ContactRequest;
import com.lacrisalide.service.EmailService;
import jakarta.validation.Valid;
import java.util.Map;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/email")
@RequiredArgsConstructor
public class EmailController {

 private final EmailService emailService;

 @PostMapping("/contatti")
 public ResponseEntity<Map<String, Object>> sendContactEmail(@Valid @RequestBody ContactRequest request) {
  boolean delivered = emailService.sendContactEmail(request);
  return ResponseEntity.ok(
   Map.of(
    "success", true,
    "delivered", delivered,
    "message", delivered
     ? "Email inviata con successo. Ti contatteremo presto!"
     : "Richiesta ricevuta correttamente. L'invio email non è configurato in questo ambiente."
   )
  );
 }
}

File: backend/Controllers/EmailController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using SoccerDreamFermana.Backend.Dtos;
using SoccerDreamFermana.Backend.Services;

namespace SoccerDreamFermana.Backend.Controllers;

[ApiController]
[Route("api/email")]
public sealed class EmailController(EmailService emailService) : ControllerBase
{
    [HttpGet("status")]
    [Authorize(Roles = "ADMIN")]
    public IActionResult GetStatus()
    {
        return Ok(emailService.GetConfigurationStatus());
    }

    [HttpPost("contatti")]
    [AllowAnonymous]
    public async Task<IActionResult> SendContactEmail(ContactRequest request, CancellationToken cancellationToken)
    {
        var result = await emailService.SendContactEmailAsync(request, cancellationToken);
        var delivered = result == ContactEmailResult.Sent;

        return Ok(new
        {
            success = true,
            delivered,
            message = result switch
            {
                ContactEmailResult.Sent => "Email inviata con successo. Ti contatteremo presto!",
                ContactEmailResult.NotConfigured => "Richiesta ricevuta correttamente. L'invio email non e configurato in questo ambiente.",
                _ => "Richiesta ricevuta correttamente, ma l'invio email non e riuscito. Riprova piu tardi."
            }
        });
    }
}
